src/internal/observable/dom/animationFrames.ts

```diff
@@ -86,15 +86,30 @@ export function animationFrames(timestampProvider?: TimestampProvider) {
  * @param timestampProvider The timestamp provider to use to create the observable
  */
 function animationFramesFactory(timestampProvider?: TimestampProvider) {
-  return new Observable<{timestamp: number, elapsed: number}>(subscriber => {
+  return new Observable<{ timestamp: number, elapsed: number }>(subscriber => {
     let id: number;
-    let start: DOMHighResTimeStamp | number | null = null;
+    // If no timestamp provider is specified, use performance.now() - as it
+    // will return timestamps 'compatible' with those passed to the run
+    // callback and won't be affected by NTP adjustments, etc.
+    const provider = timestampProvider || performance;
+    // Capture the start time upon subscription, as the run callback can remain
+    // queued for a considerable period of time and the elapsed time should
+    // represent the time elapsed since subscription - not the time since the
+    // first rendered animation frame.
+    const start = provider.now();
     const run = (timestamp: DOMHighResTimeStamp | number) => {
```

Comment:
Note that this still doesn't really solve the problem that this is untestable in TestScheduler run mode, which automatically patches our schedulers.

Reply:
I think the solution here is going to be creating a lightweight wrapper around rAF, such that we can override it in `run` mode. That's what the animationFrameScheduler is _supposed_ to be, but it's inadequate, because it doesn't provide the timestamp, and our schedulers in general are perhaps a bit overkill for this use case.

Something like this should suffice:

```ts
export const rAF = {
   schedule(callback: (timestamp: number) => void): Subscription {
     const id = this.impl(callback);
     return new Subscription(() => cancelAnimationFrame(id));
   },
   impl: requestAnimationFrame,
}
```

Then, in our `run` mode, we just need to pull that in and patch the `impl` to override it's behavior where we are using it.

We could also just monkey patch rAF directly, but that has other implications I'm not keen on. 

We also need some sort of way to declare when animation frames should be fired in a test. Probably another, idempotent helper that we add to the context passed to `TestScheduler.run`. like:

```
rxTest.run(({ cold, expectObservable, animationFrames }) => {
  animationFrames('-----x----x----x----x---x-----');
  // rest of the test here
});
```